### PR TITLE
All variable changes are now in one place without their traumatic his…

### PR DESCRIPTION
**Pull request by adanter**
…tory of having demo merged into them

### CODE REVIEW

File: Project/Assets/Editor/Twine Importer/TwineJsonParser.cs [hunk 1]
```diff
@@ -2,167 +2,240 @@
...
+    }
+
+    /// <summary>
+    /// Returns the text of a node without links or variable expressions
+    /// </summary>
+    /// <returns>The content without children atached.</returns>
+    /// <param name="content">Content with children attached.</param>
+    public static string GetVisibleText(string content)
+    {
+        Regex invisibleTextRegex = new Regex("(\\[\\[([^\\]]*)\\]\\])|(\\(\\([^)]*\\)\\))");
```
Comment (yinghj) [inline Project/Assets/Editor/Twine Importer/TwineJsonParser.cs]: Commenting the regex. Same as above.

File: Project/Assets/Prairie/Framework/Script/Twine/TwineVariables.cs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    public static TwineVariables GetVariableObject()
+    {
+        if (instance == null)
+        {
+            instance = new TwineVariables();
+        }
+        return instance;
+    }
+
+}
```
Comment (yinghj) [inline Project/Assets/Prairie/Framework/Script/Twine/TwineVariables.cs]: Add a newline at the end of file.

File: Project/Assets/Editor/Twine Importer/TwineJsonParser.cs [hunk 3]
```diff
@@ -2,167 +2,240 @@
...
+
+    /// <summary>
+    /// Parses the text of a twine node to find the variable expressions 
+    /// (marked with double parentheses) within
+    /// </summary>
+    /// <param name="text">Node text with all links and variable expressions</param>
+    /// <returns>Variable expressions minus parentheses</returns>
+    public static string[] GetVariableExpressions(string text)
+    {
+        Regex expressionRegex = new Regex("\\(\\([^)]*\\)\\)");
```
Comment (yinghj) [inline Project/Assets/Editor/Twine Importer/TwineJsonParser.cs:127]: Can you add some comments with examples maybe to indicate what kind of expressions are this is trying to parse? Thank you!

File: Project/Assets/Editor/Twine Importer/TwineJsonParser.cs [hunk 4]
```diff
@@ -2,167 +2,240 @@
...
+
+    /// <summary>
+    /// Sets the 
+    /// </summary>
+    /// <param name="expressions">Node's variable expressions</param>
+    /// <param name="node">Twine node</param>
+    /// <returns>True, unless something goes wrong</returns>
+    public static bool ActivateVariableExpressions(string[] expressions, TwineNode node)
+    {
+        Regex variableRegex = new Regex("\\$\\w*");
```
Comment (yinghj) [inline Project/Assets/Editor/Twine Importer/TwineJsonParser.cs]: Commenting the regex. Same as above.

File: Project/Assets/Editor/Twine Importer/TwineJsonParser.cs (hunk 1 above)
Comment (yinghj) [inline Project/Assets/Editor/Twine Importer/TwineJsonParser.cs]: Commenting the regex. Same as above.

**Review by yinghj — CHANGES_REQUESTED** [review]

**Review by Charlieranderson — APPROVED** [review]
Chelsea's changes have been implemented, looks good, we should merge asap

**Review by tedward111 — APPROVED** [review]

**Pull request merged**